sdk/core/azure_core_amqp/src/value.rs

```diff
@@ -18,11 +20,40 @@ use azure_core::Result;
 #[derive(Debug, PartialEq, Clone, Default, Eq)]
 pub struct AmqpSymbol(pub String);
 
+impl PartialEq<str> for AmqpSymbol {
```

Comment:
Gripe: I really wish all this was symmetric.

Technically, if you control the order in which comparisons are done (rather, the lvalue and rvalue), you don't need all these; however, if the goal is to be flexible for callers this works.